core/checkcloudconnection.cpp

```diff
@@ -56,13 +56,13 @@ bool CheckCloudConnection::checkServer()
 			}
 		} else if (seconds < prefs.cloud_timeout) {
 			QString text = tr("Waiting for cloud connection (%n second(s) passed)", "", seconds);
-			git_storage_update_progress(qPrintable(text));
+			git_storage_update_progress(qUtf8Printable(text));
```

Comment:
since we are explicitly using Qt with utf8 strings, isn't this a redundant change?

Reply:
Oh, I didn't realize that was the case:
```
#ifdef Q_OS_WIN
        QTextCodec::setCodecForLocale(QTextCodec::codecForMib(106));
#endif
```
That means we're using UTF-8 filenames on Windows?

So we could just as well turn all `qUtf8Printable()`s into `qPrintable()`s. The former executes a few instructions less (certainly not measurable). The latter is a few bytes less source code. I think it should be consistent, but apart from that I have no preference of one over the other.

Reply:
if you look closely you see all the awkward wchar16 machinations we do for Windows filenames. They are the exception. @neolit123 is there anything else we do in wchar16 on Windows?

Reply:
@bstoeger 

> So we could just as well turn all qUtf8Printable()s into qPrintable()

it kind of makes sense, unless you really want something to be independent of the locale and be 100% in UTF-8. so a search and replace is not a good idea here. AFAIK, we can pretty much use the UTF-8 codec on all platforms, so unless i'm missing something this could be a patch:

```
- #ifdef Q_OS_WIN
        QTextCodec::setCodecForLocale(QTextCodec::codecForMib(106));
- #endif
```

> That means we're using UTF-8 filenames on Windows?

classes like `QDir` and `QFile` handle that transparently for us using `QString` and there is no need to set:
`QTextCodec::setCodecForLocale(QTextCodec::codecForMib(106));` explicitly on Windows to be able to interact with the filesystem.

everything in Windows is UTF-16 (unless it's ANSI) and if you want to make WINAPI calls you probably need to provide means to translate your encoding to UTF-16. also the standard library calls have their `wchar` alternatives (like `_wfopen()`). UTF-16 was probably the worst decision they have every made which is mostly due to the fact that UTF-8 was not a primary choice back in the day. Windows is stuck with a `A` / `W` versions for most API calls.

some of that can be seen in `core/windows.c`, e.g. `utf8_to_utf16()`.